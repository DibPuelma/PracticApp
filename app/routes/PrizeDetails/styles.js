import {
  StyleSheet,
  Dimensions
} from 'react-native';

module.exports = StyleSheet.create({
  background: {
    justifyContent: 'center',
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  },
  container: {
    height: 400,
    marginLeft: 20,
    marginRight: 20,
    elevation: 3,
    alignItems: 'center',
    backgroundColor: '#E6E6E6',
    borderColor: '#C6C6C6',
    borderWidth: 1
  },
  logo: {
    width: 80,
    height: 80,
    margin: 5
  },
  storeName: {
    marginBottom: 5,
    fontSize: 25,
    fontWeight: 'bold',
    textAlign: 'right'
  },
  mediumText: {
    textAlign: 'center',
    fontSize: 20,
    marginBottom: 15
  },
  normalText: {
    textAlign: 'center',
    fontSize: 16,
    marginBottom: 20
  },
  code: {
    fontSize: 50,
    textAlign: 'center'
  }
})
