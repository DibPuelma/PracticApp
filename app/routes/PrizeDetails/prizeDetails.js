import React, { Component } from 'react';
import {
  Image,
  Text,
  View
} from 'react-native';

import backButtonHandler from '../../lib/backButtonHandler';
import styles from './styles';

export default class PrizeDetails extends Component {
  constructor(props){
    super(props);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
    this._backToPrevious = this._backToPrevious.bind(this);
  }

  componentWillMount(){
    this.singletonBackButtonHandler.addBackEvent(this._backToPrevious);
  }

  componentWillUnmount() {
    this.singletonBackButtonHandler.removeBackEvent(this._backToPrevious);
  }

  _backToPrevious() {
    this.props.navigator.pop();
    return true; // This is important to prevent multiple calls
  }

  render(){
    return(
      <View style={styles.background}>
      <View style={styles.container}>
      <Image style={styles.logo} source={{uri:this.props.prizeData.logo}} />
      <Text style={styles.storeName}> {this.props.prizeData.store}</Text>
      <Text style={styles.mediumText}> ¡Felicitaciones! Has ganado {this.props.prizeData.prize} pesos en productos {this.props.prizeData.store}.</Text>
      <Text style={styles.normalText}> Para canjearlos anda a cualquier tienda de la marca y muestra el código que aparece a continuación a cualquier vendedor</Text>
      <Text style={styles.code}> {this.props.prizeData.code} </Text>
      </View>
      </View>
    );
  }
}
