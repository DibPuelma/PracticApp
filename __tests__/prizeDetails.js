import 'react-native';
import React from 'react';
import PrizeDetails from '../app/routes/PrizeDetails/prizeDetails'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

prizeData = {
  "prize":"$100.000",
  "store":"Americanino",
  "logo":"http://www.borde.cl/wp-content/uploads/2005/11/Americanino-2.png",
  "code":"187297"
}
it('renders correctly', () => {
  const tree = renderer.create(
    <PrizeDetails prizeData={prizeData}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
