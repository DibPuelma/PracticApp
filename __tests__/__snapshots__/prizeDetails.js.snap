exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E6E6E6",
        "borderColor": "#C6C6C6",
        "borderWidth": 1,
        "elevation": 3,
        "height": 400,
        "marginLeft": 20,
        "marginRight": 20,
      }
    }>
    <Image
      source={
        Object {
          "uri": "http://www.borde.cl/wp-content/uploads/2005/11/Americanino-2.png",
        }
      }
      style={
        Object {
          "height": 80,
          "margin": 5,
          "width": 80,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
          "marginBottom": 5,
          "textAlign": "right",
        }
      }>
       
      Americanino
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 15,
          "textAlign": "center",
        }
      }>
       ¡Felicitaciones! Has ganado 
      $100.000
       pesos en productos 
      Americanino
      .
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }>
       Para canjearlos anda a cualquier tienda de la marca y muestra el código que aparece a continuación a cualquier vendedor
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 50,
          "textAlign": "center",
        }
      }>
       
      187297
       
    </Text>
  </View>
</View>
`;
